proxy/src/main/java/net/md_5/bungee/BungeeBossBar.java

```diff
@@ -0,0 +1,284 @@
+{
+
+    @Getter
+    private BaseComponent[] title;
+    @Getter
+    private BarColor color;
+    @Getter
+    private BarStyle style;
+    @Getter
+    private float progress;
+    @Getter
+    private boolean visible;
+
+    private Set<ProxiedPlayer> players;
+    private EnumSet<BarFlag> flags;
+
+    @ToString.Exclude
+    private UUID uuid;
+
+    public BungeeBossBar(BaseComponent[] title, BarColor color, BarStyle style, float progress)
+    {
+        this.title = Preconditions.checkNotNull( title, "title" );
+        this.color = Preconditions.checkNotNull( color, "color" );
+        this.style = Preconditions.checkNotNull( style, "style" );
+        Preconditions.checkArgument( 0 <= progress && progress <= 1, "Progress may not be lower than 0 or greater than 1" );
+        this.progress = progress;
+        this.visible = true;
+        this.players = new HashSet<>();
+        this.flags = EnumSet.noneOf( BarFlag.class );
+        this.uuid = UUID.randomUUID();
+    }
+
+    @Override
+    public boolean canBeAdded(ProxiedPlayer player)
+    {
+        return player.getPendingConnection().getVersion() >= ProtocolConstants.MINECRAFT_1_9;
+    }
+
+    @Override
+    public void addPlayer(ProxiedPlayer player)
+    {
+        Preconditions.checkNotNull( player, "player" );
+        if ( !canBeAdded( player ) )
+        {
+            return;
+        }
+        players.add( player );
+        if ( player.isConnected() && visible )
+        {
+            sendPacket( player, addPacket() );
+        }
+    }
+
+    @Override
+    public void addPlayers(Iterable<ProxiedPlayer> players)
+    {
+        Preconditions.checkNotNull( players, "players" );
+        for ( ProxiedPlayer player : players )
+        {
+            addPlayer( player );
+        }
+    }
+
+    @Override
+    public void removePlayer(ProxiedPlayer player)
+    {
+        Preconditions.checkNotNull( player, "player" );
+        players.remove( player );
+        if ( player.isConnected() && visible )
+        {
+            sendPacket( player, removePacket() );
+        }
+    }
+
+    @Override
+    public void removePlayers(Iterable<ProxiedPlayer> players)
+    {
+        Preconditions.checkNotNull( players, "players" );
+        for ( ProxiedPlayer player : players )
+        {
+            removePlayer( player );
+        }
+    }
+
+    @Override
+    public void removeAllPlayers()
+    {
+        for ( ProxiedPlayer player : players )
+        {
+            sendPacket( player, removePacket() );
+        }
+        players.clear();
+    }
+
+    @Override
+    public Collection<ProxiedPlayer> getPlayers()
+    {
+        return ImmutableList.copyOf( players );
+    }
+
+    @Override
+    public void setTitle(BaseComponent[] title)
+    {
+        this.title = Preconditions.checkNotNull( title, "title" );
+        if ( visible )
+        {
+            BossBar packet = new BossBar( uuid, 3 );
+            packet.setTitle( ComponentSerializer.toString( title ) );
```

Comment:
Code consistency: 3 packets are created in their own methods, so this one should be too.

Reply:
There's a reason why 3 packets are being created in their own methods.

Reply:
So what is the reason?

Reply:
The packets that have their own methods have their own methods to reduce code duplicates. The packets that doesn't have their own methods doesn't have their own methods because it is unnecessary.